src/agent/adu_core_interface/src/agent_workflow.c

```diff
@@ -676,6 +698,11 @@ void ADUC_Workflow_HandleUpdateAction(ADUC_WorkflowData* workflowData)
 
     if (entry->Action == ADUCITF_UpdateAction_Download)
     {
+        // Generate workflowId when we start downloading.
+        GenerateUniqueId(
+            workflowData->WorkflowId, sizeof(workflowData->WorkflowId) / sizeof(workflowData->WorkflowId[0]));
```

Comment:
>sizeof(workflowData->WorkflowId) / sizeof(workflowData->WorkflowId[0]) [](http://example.com/codeflow?start=38&length=70)

You could use ARRAY_SIZE macro here..

e.g. 

ARRAY_SIZE(workflowData->WorkflowId)